elasticsearch_dsl/_async/search.py

```diff
@@ -158,16 +164,20 @@ async def iterate(self, keep_alive="1m"):
                     yield hit
                 if len(r.hits) == 0:
                     break
-                s = r.search_after()
+                s = s.search_after()
```

Comment:
This does not change the behavior, right?

Reply:
No, the type hint in the search call is more specific than the one in the response, so it works better for type checking. But they do the same thing.